src/features/sodax/hooks/useSodaxSwap.tsx

```
@@ -0,0 +1,451 @@
+    SodaxSwapStep.IDLE,
+  );
+  const [fillStatus, setFillStatus] = useState<SodaxSubmitStatus | null>(null);
+  const [error, setError] = useState<SodaxSwapError | null>(null);
+  const [result, setResult] = useState<SodaxSwapResult | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const abortRef = useRef(false);
+
+  // Callbacks live in a ref so a fresh options literal from the caller does
+  // not rebuild every callback (and everything downstream) on each render.
+  const optionsRef = useRef(options);
+  optionsRef.current = options;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/features/sodax/hooks/useSodaxSwap.tsx:151:3
  149 |   // not rebuild every callback (and everything downstream) on each render.
  150 |   const optionsRef = useRef(options);
> 151 |   optionsRef.current = options;
      |   ^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  152 |
  153 |   // Stop polling when the component unmounts (route change, error boundary).
  154 |   useEffect(() => {

(react-hooks/refs)